Guides/Error-Handling.md

```diff
@@ -0,0 +1,143 @@
+# Swift Driver Error Handling Guide
+
+## Index
+* [Error Types](#error-types)
+    * [Server Errors](#server-errors)
+    * [User Errors](#user-errors)
+    * [Runtime Errors](#runtime-errors)
+    * [Encoding/Decoding Errors](#encoding-decoding-errors)
+* [Examples](#the-code)
+    * [All Errors](#handling-any-error-thrown-by-the-driver)
+    * [CommandError](#handling-a-commanderror)
+    * [WriteError](#handling-a-writeerror)
+    * [BulkWriteError](#handling-a-bulkwriteerror)
+* [See Also](#see-also)
+
+## Error Types
+The driver uses errors to communicate that an operation failed, an assumption wasn't met, or that the user did something incorrectly. Applications that use the driver can in turn catch these errors and respond appropriately without crashing or resulting in an otherwise inconsistent state. To correctly model the different sources of errors, the driver defines three separate types of errors (`ServerError`, `UserError`, `RuntimeError`), each of which conforms to the `MongoError` protocol. These errors are defined in `MongoError.swift` and are outlined here. The documentation for every public function that throws lists some of the errors that could possibly be thrown and the reasons they might be. The errors listed there are not comprehensive but will generally cover the most common cases.
+
+**Error Labels:** Some types of errors may contain more specific information describing the context in which they occured. This information is conveyed through the usage of `errorLabels`. Specifically, any server error or connection related error may contain labels. They are primarily used for classifying errors that occur when performing transactions, so at this point they will largely be unused.
```

Comment:
maybe leave off the last sentence, we will probably forget to remove it whenever we get to adding 4.0 support... lol. or maybe add a ticket to 4.0 epic telling us to update this guide to talk more about transaction errors? if you do leave it I would suggest stating explicitly why this is the case (no transactions support).